src/pages/index.js

```diff
@@ -76,21 +93,118 @@ const popupWithImage = new PopupWithImage("#image-popup");
 popupWithImage.setEventListeners();
 
 //Section
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      renderCard(cardData);
-    },
-  },
-  "cards__list"
-);
+let section;
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardEl = renderCard(cardData);
+          section.addItem(cardEl);
+        },
+      },
+      "cards__list"
+    );
+    section.renderItems();
+  })
+
+  .catch((error) => {
+    console.error(error);
+  });
+
+api
+.getUserInfo()
+  .then((user) => {
+    userInfo.setUserInfo(user.name, user.about);
+    userInfo.setAvatar(user.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-section.renderItems();
 //UserInfo
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.description);
-  editPopup.close();
+  editPopup.setLoading(true);
+  api.editUserInfo(inputValues).then(() => {
+    userInfo.setUserInfo(inputValues.name, inputValues.description);
+    editPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    editPopup.setLoading(false);
+  });
+}
+
+//EditAvatar
+
+const updateAvatarForm = new PopupWithForm("#profile-image-modal", handleAvatarSubmit);
+function handleAvatarSubmit(avatar) {
+  updateAvatarForm.setLoading(true);
+  api
+    .updateAvatar(avatar.link)
+    .then((user) => {
+      userInfo.setAvatar(user.avatar);
+      updateAvatarForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      updateAvatarForm.setLoading(false);
+    });
 }
+
+const editPencilIcon = document.querySelector("#avatar-edit-button");
+
+editPencilIcon.addEventListener("click", () => {
+  updateAvatarForm.open();
+});
+
+updateAvatarForm.setEventListeners();
+//PopupWithConfirmation
+
+const deleteCardPopup = new PopupWithConfirmation("#delete-card-modal");
+
+function handleDeleteButton(cardID, card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+  });
+}
+
+deleteCardPopup.setEventListeners();
+
+
+
+//addCard
+
+const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+
+// function renderCard(cardData) {
+//   const card = createCard(cardData);
+// }
+
+function handleAddCardFormSubmit(inputValues) {
+  cardPopup.setLoading(true);
+  api.addNewCard(inputValues).then((card) => {
+    const cardEl = renderCard(card);
+    section.addItem(cardEl);
+    cardPopup.close();
+    cardPopup.setLoading(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.